app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt

```diff
@@ -103,14 +103,10 @@ fun ProfileScreen(viewModel: ProfileViewModel) {
 		)
 	}
 	if (showProfileUpdateBottomSheet) {
-		userName = viewModel.userName.value
 		ProfileUpdateBottomSheet(
-			userName = userName,
-			updateUserName = { userName = it },
+			userName = viewModel.userName.value,
 			updateName = { viewModel.updateName(it) },
 			updateVisibility = { showProfileUpdateBottomSheet = it },
-			validateName = { viewModel.validatingName(it).isValid() },
-			getInvalidNameErrorMessageId = { viewModel.validatingName(it).errorMessageId },
```

Comment:
The main issue here is rooted in `ValidationStatus`, for some reason `ValidText` has set `errorMessageId` to `TextIsValid` which is an empty string, in resources. So instead of hiding "error view" we set the text to empty string. Im most of the cases, the view shouldn't visible, not just with no text, because it can occupy some space on the screen due to padding, would that be an expected behavior?

1.I would recommend to set `errorMessageId` for `ValidationStatus` to `null` - one PR

2. view model should keep `TextValidator` and expose only `errorMessageId`. Parameter `validateName` and `getInvalidNameErrorMessageId` can be merged into then.

Reply:
Done, but the view model doesn't expose errorMessageId. I'm returning it from TextValidator and it is handled by the bottom sheet state